File: CS/MyProgressIndicator.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.XtraEditors;
using DevExpress.Services;
using System.Windows.Forms;
using DevExpress.Xpf.Editors;
using System.Windows.Threading;

namespace DocumentVariablesExample
{
    #region #myprogressindicator
    class MyProgressIndicatorService : IProgressIndicationService
    {
        private ProgressBarEdit _Indicator;
        public ProgressBarEdit Indicator
        {
            get { return _Indicator; }
            set { _Indicator = value; }
        }

        public MyProgressIndicatorService(IServiceProvider provider, ProgressBarEdit indicator)
        {
            _Indicator = indicator;
        }
        
        #region IProgressIndicationService Members

        void IProgressIndicationService.Begin(string displayName, int minProgress, int maxProgress, int currentProgress)
        {
            _Indicator.Minimum = minProgress;
            _Indicator.Maximum = maxProgress;
            _Indicator.EditValue = currentProgress;
            _Indicator.Visibility = System.Windows.Visibility.Visible;
            Refresh();
        }

        void IProgressIndicationService.End()
        {
            _Indicator.Visibility = System.Windows.Visibility.Collapsed;
            Refresh();
        }

        void IProgressIndicationService.SetProgress(int currentProgress)
        {
            _Indicator.EditValue = currentProgress;
            Refresh();
        }
        #endregion

        void Refresh() {
            Action emptyDelegate = delegate() { };
            _Indicator.Dispatcher.Invoke(DispatcherPriority.Render, emptyDelegate);
        }
    }
#endregion #myprogressindicatorsindicator
}

File: VB/MyProgressIndicator.vb
Imports System
Imports DevExpress.Services
Imports DevExpress.Xpf.Editors
Imports System.Windows.Threading

Namespace DocumentVariablesExample

'#Region "#myprogressindicator"
    Friend Class MyProgressIndicatorService
        Implements IProgressIndicationService

        Private _Indicator As ProgressBarEdit

        Public Property Indicator As ProgressBarEdit
            Get
                Return _Indicator
            End Get

            Set(ByVal value As ProgressBarEdit)
                _Indicator = value
            End Set
        End Property

        Public Sub New(ByVal provider As IServiceProvider, ByVal indicator As ProgressBarEdit)
            _Indicator = indicator
        End Sub

'#Region "IProgressIndicationService Members"
        Private Sub Begin(ByVal displayName As String, ByVal minProgress As Integer, ByVal maxProgress As Integer, ByVal currentProgress As Integer) Implements IProgressIndicationService.Begin
            _Indicator.Minimum = minProgress
            _Indicator.Maximum = maxProgress
            _Indicator.EditValue = currentProgress
            _Indicator.Visibility = Windows.Visibility.Visible
            Refresh()
        End Sub

        Private Sub [End]() Implements IProgressIndicationService.End
            _Indicator.Visibility = Windows.Visibility.Collapsed
            Refresh()
        End Sub

        Private Sub SetProgress(ByVal currentProgress As Integer) Implements IProgressIndicationService.SetProgress
            _Indicator.EditValue = currentProgress
            Refresh()
        End Sub

'#End Region
        Private Sub Refresh()
            Dim emptyDelegate As Action = Sub()
            End Sub
            _Indicator.Dispatcher.Invoke(DispatcherPriority.Render, emptyDelegate)
        End Sub
    End Class
'#End Region  ' #myprogressindicatorsindicator
End Namespace
